src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.validation.DataValidator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private final DataValidator validator;
+
+    public DataConverterImpl(DataValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        validator.validate(inputData);
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+
+        for (String line : inputData.subList(1, inputData.size())) {
+            String[] partsOfData = line.split(",");
```

Comment:
No validation or error handling for the split data. It is assumed that each line will correctly split into exactly three parts. If any line doesn't conform to this, it will throw an ArrayIndexOutOfBoundsException.

Reply:
I added a check to ensure partsOfData has exactly three elements. If not, an IllegalArgumentException is thrown.